G/GMP/GMP@6.2.1/build_tarballs.jl

```diff
@@ -0,0 +1,60 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+# Collection of sources required to build GMP
+name = "GMP"
+version = v"6.2.1"
+
+sources = [
+    ArchiveSource("https://gmplib.org/download/gmp/gmp-$(version).tar.bz2",
+                  "eae9326beb4158c386e39a356818031bd28f3124cf915f8c5b1dc4c7a36b4d7c"),
+    DirectorySource("./bundled"; follow_symlinks=true),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/gmp-*
+
+# Include Julia-carried patches
+for f in $WORKSPACE/srcdir/patches/*.patch; do
+    echo "Applying patch ${f}"
+    atomic_patch -p1 ${f}
+done
+
+flags=(--enable-cxx --enable-shared --disable-static)
+
+# On x86_64 architectures, build fat binary
+if [[ ${proc_family} == intel ]]; then
+    flags+=(--enable-fat)
+fi
+
+autoreconf
+./configure --prefix=$prefix --build=${MACHTYPE} --host=${target} ${flags[@]}
+
+make -j${nproc}
+make install
+
+# On Windows, we need to make sure that the non-versioned dll names exist too
+if [[ ${target} == *mingw* ]]; then
+    cp -v ${libdir}/libgmp-*.dll "${libdir}/libgmp.dll"
+    cp -v ${libdir}/libgmpxx-*.dll "${libdir}/libgmpxx.dll"
+fi
+
+# GMP is dual-licensed, install all license files
+install_license COPYING*
+"""
+
+# We enable experimental platforms as this is a core Julia dependency
+platforms = expand_cxxstring_abis(supported_platforms(;experimental=true))
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libgmp", :libgmp),
+    LibraryProduct("libgmpxx", :libgmpxx),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = []
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"6", julia_compat="1.6")
```

Comment:
Is this going to be in Julia v1.6?

Reply:
Unlikely. It is a patch release though. I can bump to 1.7 and we can always update and make a new build if needed on 1.6

Reply:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"6", julia_compat="1.7")
```